Complete content for RStudio IDE docs (#12075)
rstudio/rstudio: docs/user/rstudio/ide/guide/tools/version-control.qmd

-   [Subversion](http://subversion.apache.org/)

To use version control with RStudio, first ensure that Git and/or Subversion tools are installed on your workstation (details below).

> Version control is most useful when used with a remote repository. Remote repositories are typically managed by your company or are hosted in the cloud (e.g. [Github](https://github.com/) or [GitLab](https://about.gitlab.com/)). Make sure you have credentials to access these systems. If you only want to learn how to use version control, you can manage a standalone system on your workstation but you will not be able to share code with others.

Reviewer: ```suggestion
Version control is most useful when used with a remote repository. Remote repositories are typically managed by your company or are hosted in the cloud (e.g., [Github](https://github.com/) or [GitLab](https://about.gitlab.com/)). Make sure you have credentials to access these systems. If you only want to learn how to use version control, you can manage a standalone system on your workstation but you will not be able to share code with others.
```

Author: added comma

> You should also become familiar with using [RStudio Projects](../authoring/projects.html) (which are required for version control features to be enabled).

Reviewer 2: ```suggestion
You should also become familiar with using [RStudio Projects](../code/projects.html) (which are required for version control features to be enabled).
```

Author: Changed - thanks!

> 

Reviewer 2: In the linux commands for installing git below, why `install git-core` rather than `install git`? I'm referencing this: https://git-scm.com/download/linux

Author: Great call! This was a quite old article - git-core used to be required as there was a namespace conflict with git. I've completely refactored this to the below, as those resources have detailed instructions for every platform.

> Prior to using Git with RStudio you should install it using the appropriate method for your platform as indicated at <https://git-scm.com/download/linux>.

and for subversion:

> Prior to using Subversion with RStudio you should install it using the appropriate method for your platform as indicated at <https://subversion.apache.org/packages.html>.

> 

Reviewer 2: do the mac osx (v10.8+) instructions for installing subversion need to be updated? looks like you can grab it from homebrew (https://subversion.apache.org/packages.html#osx) and I think it may have been removed from [Xcode command line tools](https://mac.install.guide/commandlinetools/8.html)

Author: Thanks! I removed the install instructions in favor of linking out to their install guides which have specific OS details.

## Getting Started

Once a Version Control system has been installed, it needs to be activated on the system by following these steps:

> 1.  Go to **Global Options** (from the Tools menu)

Reviewer: ```suggestion
1.  From the Tools menu, click **Global Options** 
```

Author: Changed order

> 2.  Press **Git/SVN**

Reviewer: If this is a button, we should change "Press" to Click.

Author: Changed to click

> 3.  Press **Enable version control interface for RStudio projects**

Reviewer: If this is a button, we should change "Press" to Click.

Author: Changed to click

> 4.  If necessary, enter the path for the Git or SVN executable where provided. A RSA key can be created or added for SSH if necessary.

Reviewer 3: should be An SSH key

We've added support for other types of SSH keys besides RSA and RSA is no longer the default

Author: Changed to SSH throughout

![](images/version-control-options.png){fig-alt="A screenshot of the Git/SVN tab in Global Options displaying an active Git executable."}

If you are using RStudio Server, by default RStudio Server will look at the PATH for the first Git executable it finds, and sets that as the default. If you have a specific version that you want to set for all users, the easiest option is to make sure it's the first available on the PATH system-wide.

Alternately, if you are running version 1.4 or above of Posit Workbench (previously RStudio Server Pro/RStudio Workbench), you can set the default Git version system-wide by setting the `git_exe_path` option in your system rstudio-prefs.json file. See here for more:\
<https://docs.posit.co/ide/server-pro/r-sessions.html#customizing-session-settings>\

> <https://docs.posit.co/ide/server-pro/session-user-settings.html>

Reviewer 2: these two links didn't load for me. are they live yet?

Author: They are not but will be by the release of the IDE guide

> RStudio's version control features are tied to the use of [Projects](../code/projects.html) (which are a way of dividing work into multiple contexts, each with their own working directory).The steps required to use version control with a project vary depending on whether the project is new or existing as well as whether it is already under version control.

Reviewer 2: ```suggestion
RStudio's version control features are tied to the use of [Projects](../code/projects.html) (which are a way of dividing work into multiple contexts, each with their own working directory). The steps required to use version control with a project vary depending on whether the project is new or existing as well as whether it is already under version control.
```

### Integrations

Once version control such as Git has been enabled for the current RStudio Project, a few visual elements are added, specifically the **Git/Version control dropdown** and the **Git/Version Control tab**.

![](images/rstudio-vcs-pane-labeled.png){fig-alt="A screenshot of the RStudio interface, highlighting the Git tab in the Environments pane."}

The **Git menu dropdown** provides Commit, Pull/Push Branch, History and Project Setup.

![](images/git-drop-down.png){width="350" fig-alt="A screenshot of the git dropdown."}

The **Git tab** within the **Environment pane** provides a number of useful commands such as "Diff", "Commit", the Push/Pull arrows, View history 'clock', or selecting a specific branch.

![](images/git-tab.png){fig-alt="A screenshot of the Git tab in RStudio, showing a few files that can be added to staging."}

> Once you are finished with your local work and are ready to Commit your changes, selecting **Commit** in either the **Git dropdown menu** or **Git tab** opens the **Review Changes popup** to select files for staging, add a commit message, and eventually push the commit.

Reviewer: ```suggestion
Once you are finished with your local work and are ready to Commit your changes, selecting **Commit** in either the **Git** drop-down menu or **Git** tab opens the **Review Changes** pop-up to select files for staging, add a commit message, and eventually push the commit.
```

Author: Changed

> Once you are finished with your local work and are ready to Commit your changes, selecting **Commit** in either the **Git** drop-down menu** or **Git** tab opens the **Review Changes** pop-up to select files for staging, add a commit message, and eventually push the commit.

Reviewer 2: ```suggestion
Once you are finished with your local work and are ready to Commit your changes, selecting **Commit** in either the **Git** drop-down menu or **Git** tab opens the **Review Changes** pop-up to select files for staging, add a commit message, and eventually push the commit.
```

Author: Removed trailing `**`
